File: k9mail/src/test/java/com/fsck/k9/crypto/OpenPgpApiHelperTest.java
package com.fsck.k9.crypto;


import com.fsck.k9.Identity;
import org.junit.Test;

import static org.junit.Assert.assertEquals;


public class OpenPgpApiHelperTest {

    @Test
    public void buildUserId_withName_shouldCreateOpenPgpAccountName() {
        Identity identity = new Identity();
        identity.setEmail("user@domain.com");
        identity.setName("Name");

        String result = OpenPgpApiHelper.buildUserId(identity);

        assertEquals("Name <user@domain.com>", result);
    }

    @Test
    public void buildUserId_withoutName_shouldCreateOpenPgpAccountName() {
        Identity identity = new Identity();
        identity.setEmail("user@domain.com");

        String result = OpenPgpApiHelper.buildUserId(identity);

        assertEquals("<user@domain.com>", result);
    }

}

File: legacy/core/src/test/java/com/fsck/k9/crypto/OpenPgpApiHelperTest.kt
package com.fsck.k9.crypto

import app.k9mail.legacy.account.Identity
import assertk.assertThat
import assertk.assertions.isEqualTo
import org.junit.Test

class OpenPgpApiHelperTest {

    @Test
    fun buildUserId_withName_shouldCreateOpenPgpAccountName() {
        val identity = Identity(
            email = "user@domain.com",
            name = "Name",
        )

        val result = OpenPgpApiHelper.buildUserId(identity)

        assertThat(result).isEqualTo("Name <user@domain.com>")
    }

    @Test
    fun buildUserId_withoutName_shouldCreateOpenPgpAccountName() {
        val identity = Identity(
            email = "user@domain.com",
        )

        val result = OpenPgpApiHelper.buildUserId(identity)

        assertThat(result).isEqualTo("<user@domain.com>")
    }
}
